url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/rule/Rule.java

```diff
@@ -0,0 +1,173 @@
+/*   Copyright 2013 Juan Rada-Vilela
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.fuzzylite.rule;
+
+import com.fuzzylite.Engine;
+import com.fuzzylite.Op;
+import static com.fuzzylite.Op.str;
+import com.fuzzylite.norm.SNorm;
+import com.fuzzylite.norm.TNorm;
+import java.util.StringTokenizer;
+
+/**
+ *
+ * @author jcrada
+ */
+public class Rule {
+
+    public static final String FL_IF = "if";
+    public static final String FL_IS = "is";
+    public static final String FL_EQUALS = "=";
+    public static final String FL_THEN = "then";
+    public static final String FL_AND = "and";
+    public static final String FL_OR = "or";
+    public static final String FL_WITH = "with";
+    protected Antecedent antecedent;
+    protected Consequent consequent;
+    protected double weight = 1.0;
+    protected String text;
+
+    public Rule() {
+    }
+
+    public double activationDegree(TNorm conjunction, SNorm disjunction) {
+        return weight * this.antecedent.activationDegree(conjunction, disjunction);
+    }
+
+    public void activate(double activationDegree, TNorm activation) {
+        this.consequent.modify(activationDegree, activation);
+    }
+
+    public static Rule parse(String rule, Engine engine) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 45 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 108-120 into a separate method.